File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while (l <= r) {
            int mid = l + floor((r - l) / 2);
            if (target < nums[mid]) {
                r = --mid;
            }
            else if (target > nums[mid]) {
                l = ++mid;
            }
            else return mid;
        };
        return -1;
    };
};

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums) == 1:
            if target in nums: return 0
            else: return -1
        
        mid = math.floor(len(nums)/2)
        buffer = mid

        for count in range(0, math.floor(len(nums)/2) + 1):
            if nums[mid] == target: return mid
            if nums[mid] > target: mid = math.floor(mid/2)
            else:
                buffer = max(math.floor(buffer/2), 1)
                mid = min(mid + buffer, len(nums)-1)
        return -1